### ISSUE TITLE
Batch size is structurally locked at 2 by the training loop

### ISSUE THREAD

**Issue by abuzarmahmood** [pulakatlab/autoslide#122]
Part of #116.

## The defect

The training loop indexes the batch positionally:

https://github.com/pulakatlab/autoslide/blob/main/autoslide/src/pipeline/model/training_utils.py#L139-L140

```python
imgs = [dt[0][0].to(device), dt[1][0].to(device)]
targ = [dt[0][1], dt[1][1]]
```

`custom_collate` is the identity function, and `batch_size = 2` is hardcoded in
`create_dataloaders` with `drop_last=True`. Because the loop literally reads
elements 0 and 1, **the batch size cannot be changed without editing the
training loop** — raising `batch_size` would silently discard the rest of every
batch. The same pattern appears in the validation loop.

`split_train_val` even rounds the training set to an even number to accommodate
this:

```python
num = num if num % 2 == 0 else num + 1
```

## Why it matters

Batch size 2 is the dominant source of gradient noise in the current setup, and
it forces `lr=0.005` to stay at the tutorial default rather than being tuned.
At 706x706, a batch of 8 fits comfortably on most GPUs.

## Suggested fix

```python
def custom_collate(batch):
    return tuple(zip(*batch))

# in the loop
imgs = [img.to(device) for img in dt[0]]
targets = [{k: v.to(device) for k, v in t.items()} for t in dt[1]]
```

Then `batch_size` becomes a real knob, `drop_last=True` can go, and the
even-number rounding in `split_train_val` can be removed. Scale the learning
rate with the batch size when benchmarking.
